src/site/topic7.rst

```diff
@@ -164,16 +164,86 @@ Deleting from the End
 Node Implementation
 ===================
 
+
 .. image:: img/links_reference.png
    :width: 500 px
    :align: center
 
 * Remember, reference variables contain a reference to an object
 * The linked structure uses these references to link it together
 
+
+* The node implementation for the singly linked structure is kept simple
+    * A field to keep track of the data
+    * A field to keep track of the next/successor node
+    * Constructors
+    * Getters and setters
+
+.. code-block:: java
+    :linenos:
+
+    public class Node<T> {
+
+        private T data;
+        private Node<T> next;
+
+        public Node() {
+            this(null);
+        }
+
+        public Node(T data) {
+            this.data = data;
+            this.next = null;
+        }
+
+        public T getData() {
+            return data;
+        }
+
+        public void setData(T data) {
+            this.data = data;
+        }
+
+        public Node<T> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<T> next) {
+            this.next = next;
+        }
+    }
+
+
 Explore the Implementation
 --------------------------
 
+.. code-block:: java
+    :linenos:
+
+    // Create a Node
+    Node<Integer> head = new Node<Integer>(5);
```

Comment:
Minor, since java7 (iirc) you can save yourself some typing and verbosity by using the diamond operator.
```java
Node<Integer> head = new Node<>(5);  // no need to specify `Integer` twice
```

Reply:
I will make an issue on this as I am pretty sure I have been inconsistent with using this.